src/widget/button.rs

```diff
@@ -47,34 +45,39 @@ impl Label {
         ctx.add(self, &[])
     }
 
-    fn get_layout(&self, dwrite_factory: &directwrite::Factory) -> TextLayout {
+    fn get_layout(
+        &self,
+        rt: &mut Piet,
+        font_size: f32,
+    ) -> <<Piet as RenderContext>::Text as Text>::TextLayout {
         // TODO: caching of both the format and the layout
-        let format = TextFormat::create(&dwrite_factory)
-            .with_family("Segoe UI")
-            .with_size(15.0)
+        let font = rt
+            .text()
+            .new_font_by_name("Segoe UI", font_size)
+            .unwrap()
             .build()
             .unwrap();
-        let layout = TextLayout::create(&dwrite_factory)
-            .with_text(&self.label)
-            .with_font(&format)
-            .with_width(1e6)
-            .with_height(1e6)
+        // TODO: Layout height and width: 1e6
+        rt.text()
+            .new_text_layout(&font, &self.label)
+            .unwrap()
             .build()
-            .unwrap();
-        layout
+            .unwrap()
     }
 }
 
 impl Widget for Label {
     fn paint(&mut self, paint_ctx: &mut PaintCtx, geom: &Geometry) {
-        let text_layout = self.get_layout(paint_ctx.dwrite_factory());
-        let rt = paint_ctx.render_target();
-        let fg = SolidColorBrush::create(rt)
-            .with_color(0xf0f0ea)
-            .build()
+        let font_size = 15.0;
+        let text_layout = self.get_layout(paint_ctx.render_ctx, font_size);
+        let brush = paint_ctx.render_ctx.solid_brush(0xf0f0eaff).unwrap();
+
+        // TODO: bring back default_text_options from win-shell
```

Comment:
I think this is the wrong place for this TODO. There's already [one](https://github.com/linebender/piet/blob/7a13480b19936b020726499f81f6388545e2b807/piet-direct2d/src/lib.rs#L274) in piet-direct2d. (It should be noted that this change will regress color emoji, but it's not going to be that difficult to fix, it's basically a factor of having Windows version detection at the right place).